src/app/shared/services/dynamic-data/adapters/persistedMemory.ts

```diff
@@ -118,8 +118,14 @@ export class PersistedMemoryAdapter {
 
   public delete(flow_type: FlowTypes.FlowType, flow_name: string) {
     if (this.get(flow_type, flow_name)) {
+      const idsToDelete = [];
+      const rowHash = this.state[flow_type][flow_name];
+      for (const row_id of Object.keys(rowHash)) {
+        idsToDelete.push(`${flow_type}__${flow_name}__${row_id}`);
+      }
       delete this.state[flow_type][flow_name];
-      this.persistStateToDB();
+      // Persist deletes to DB
+      this.collection.bulkRemove(idsToDelete);
```

Comment:
nit(non-blocking)
I'm struggling to follow this a bit, could you clarify why we no longer need the `persistStateToDB` method call following delete and what the extra code does in its place?

Reply:
The function `persistStateToDB()` didn't handle deletions because it only performed a bulk _upsert_. I've now updated that with [this commit](https://github.com/IDEMSInternational/parenting-app-ui/pull/1989/commits/76f1217894d65c9a79cd677f95567ece98e00a7a) so that `persistStateToDB()` should handle deletes in general. I'm not sure that I've found the simplest solution though.